src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -24,14 +24,19 @@ def self.initialize_puppet_settings(puppet_config:, require_config:, push_settin
     # to override, and whose values are the desired values for the settings.
     # Values that are not strings are omitted, allowing for keys in the
     # HashMap from puppetserver to have true values.
-    Puppet.initialize_settings(
-        puppet_config.reduce([]) do |acc, entry|
-          acc << "--#{entry[0]}"
-          acc << entry[1] if entry[1].kind_of?(String)
-          acc
-        end,
-        require_config, push_settings_globally
-    )
+    cli_flags = puppet_config.reduce([]) do |acc, entry|
+      acc << "--#{entry[0]}"
+      acc << entry[1] if entry[1].kind_of?(String)
+      acc
+    end
+
+    # We check parameter length because Method#arity returns -1 for varargs
+    if Puppet.method(:initialize_settings).parameters.length == 3
+      Puppet.initialize_settings(cli_flags, require_config, push_settings_globally)
```

Comment:
I suppose it's too late to make changes to puppet, but we probably should've just made these optional args?

Reply:
Just to be pedantic, they are already optional, but the previous method takes 0-2 args and new one takes 0-3 args (that variable argument-ness is why arity return -1 for both signatures). But I know what you mean.

Reply:
ugh that's annoying.. I feel like having a variable number of args is supposed to prevent breakages exactly like this??

Reply:
oh wait, I see what you're saying

Reply:
we're calling it with the 3rd one here

Reply:
They are optional in [Puppet](https://github.com/puppetlabs/puppet/blob/master/lib/puppet.rb#L154), but I think the problem is when running the server with an older version of Puppet that isn't aware of the 3rd positional argument, `push_settings_globally`.

Reply:
yep 👍